api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -140,6 +141,9 @@ message DownstreamTlsContext {
   // If the client provides SNI but no such cert matched, it will decide to full scan certificates or not based on this config.
   // Defaults to false. See more details in :ref:`Multiple TLS certificates <arch_overview_ssl_cert_select>`.
   google.protobuf.BoolValue full_scan_certs_on_sni_mismatch = 9;
+
+  // Config for external TLS session cache.
+  config.tls_session_cache.v3.TlsSessionCacheConfig session_cache_service = 11;
```

Comment:
rather than a specific grpc-based external tls cache, I will prefer to use a `core.v3.TypedExtensionConfig` here and make the grpc a specific implementation.

Although I agree we could use the grpc as an adapter to handle any type back stores. But in the practices, we prefer to reduce the components that are involved in at a specific scenario/feature.
core.v3.TypedExtensionConfig empowers more possibilities. For example, local disk based cache, shared memory based cache, etc.

And then we can split this work to two independent PRs: one for the core API and code of new extension, one for the grpc implementation.

cc @ggreenway

Reply:
+1 for adding as an extension point

Reply:
This API comments still not be addressed.

Reply:
/wait